src/security-council-mgmt/SecurityCouncilManager.sol

```diff
@@ -0,0 +1,266 @@
+///         and dispatches them to all security councils on all relevant chains
+contract SecurityCouncilManager is
+    Initializable,
+    AccessControlUpgradeable,
+    ISecurityCouncilManager
+{
+    using ECDSA for bytes32;
+
+    // cohort arrays are source-of-truth for security council; the maximum 12 owners security council owners should always be equal to the
+    // sum of these two arrays (or have pending x-chain messages on their way to updating them)
+    address[] public marchCohort;
+    address[] public septemberCohort;
+
+    bytes32 public constant ELECTION_EXECUTOR_ROLE = keccak256("ELECTION_EXECUTOR");
+    bytes32 public constant MEMBER_ADDER_ROLE = keccak256("MEMBER_ADDER");
+    bytes32 public constant MEMBER_ROTATOR_ROLE = keccak256("MEMBER_ROTATOR");
+    bytes32 public constant MEMBER_REMOVER_ROLE = keccak256("MEMBER_REMOVER");
+
+    address public l1SecurityCouncilUpdateRouter;
+
+    event L1SecurityCouncilUpdateRouterSet(address indexed l1SecurityCouncilUpdateRouter);
+    event ElectionResultHandled(address[] newCohort, Cohort indexed cohort);
+    event MemberAdded(address indexed newMember, Cohort indexed cohort);
+    event MemberRemoved(address indexed member, Cohort indexed cohort);
+
+    uint256 public rotationNonce = 0;
+
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(
+        address[] memory _marchCohort,
+        address[] memory _septemberCohort,
+        Roles memory _roles,
+        address _l1SecurityCouncilUpdateRouter
+    ) external initializer {
+        marchCohort = _marchCohort;
+        septemberCohort = _septemberCohort;
+        // TODO: ensure march + september cohort = all signers?
+        _grantRole(DEFAULT_ADMIN_ROLE, _roles.admin);
+        _grantRole(ELECTION_EXECUTOR_ROLE, _roles.cohortUpdator);
+        _grantRole(MEMBER_ADDER_ROLE, _roles.memberAdder);
+        for (uint256 i = 0; i < _roles.memberRemovers.length; i++) {
+            _grantRole(MEMBER_REMOVER_ROLE, _roles.memberRemovers[i]);
+        }
+        _grantRole(MEMBER_ROTATOR_ROLE, _roles.memberRotator);
+
+        _setL1SecurityCouncilUpdateRouter(_l1SecurityCouncilUpdateRouter);
+    }
+
+    /// @notice callable only by Election Governer. Updates cohort in this contract's state and triggers dispatch.
+    /// @param _newCohort new cohort to replace existing cohort. New cohort is result of election, so should always have 6 members.
+    /// @param _cohort cohort to replace.
+    function executeElectionResult(address[] memory _newCohort, Cohort _cohort)
+        external
+        onlyRole(ELECTION_EXECUTOR_ROLE)
+    {
+        require(_newCohort.length == 6, "SecurityCouncilManager: invalid cohort length");
+        // TODO: ensure no duplicates accross cohorts; this should be enforced in nomination process. If there are duplicates, this call will revert in the Gnosis safe contract
```

Comment:
While this said if there is duplicate it will revert in gnosis, in `SecurityCouncilUpgradeExecutor.updateMembers` it prevented the revert by ignoring duplicated entries.

Reply:
updated comment